### ISSUE TITLE
Addind support for metric prefix

### ISSUE THREAD

**Issue by uncle-buzz** [petermorlion/RedStar.Amounts#3]
Hi,
Thank you for your share.
It would be a good enhancement to add support for metric prefix (https://en.wikipedia.org/wiki/Metric_prefix) automatically for all units because milli is always a 0.001 factor, as kilo is always a 1000 factor, without dependencies with the main unit it precedes, and as a prefix, the string representation and symbol will always be build the same way.

**Comment by petermorlion** [comment]
Could you provide an example? When you create a new Amount like this:

```
var a = new Amount(30, LengthUnits.MilliMeter);
```

the string representation will be "30 mm".

However, if you create one using:

```
var a = new Amount(30, 0.001 * LengthUnits.Meter);
```

you are correct, it will be "30 0.001*m

Is it in that case that you want "30 mm"?

I might be able to achieve this by scanning for matching unit definitions. I'm thinking if you want this explicit or implicit, ie:

```
new Amount(30, 0.001 * LengthUnits.Meter).ToString(); //30 mm
// or
new Amount(30, 0.001 * LengthUnits.Meter).AutoConvert().ToString(); // 30 mm
```

Do you have an opinion on that?

**Comment by uncle-buzz** [comment]
Not exactly, I don't speak about autoconvertion, 30 mm is also 0.030 m, there is no difference and 30 mm is not automatically better than 0.030 m, it depends of each application and autoconvertion won't be able to assert all cases.
What I mean is millimeter is not really a new unit, it's the same unit than meters. When you declare a new unit like meter, liter, amper, gram... millimeter, kilogram... should be automatically recognise as unit without be explicitely declared. They are same units with implicit factor.
For each unit, all the metric prefix should be recognise as a new unit also.
If  I create the meter unit, micrometer, millimeter, kilometer... should be created in same time, with correct factor, correct ToString() representation as abreviation representation.
It needs to introduce something like main units, and subunits which are a combination of metric prefix and main units.
Since factor, representation and abreviation for each prefix are constant, it would be possible to create all subunits in same time we declare a main unit.
```
[UnitDefinitionClass]
public static class LengthUnits
{
public static readonly Unit Meter = new Unit("meter", "m", SIUnitTypes.Length);
}

var a = new Amount(0.03, LengthUnits.Meter);
a.ToString(); // 0.03 m
a.ConvertedTo(Millimeter).ToString(); // 30 mm
a.ConvertedTo(Micrometer).ToString(); // 30000 µm

var b = new Amount(300, LengthUnits.NanoMeter);
b.ToString(); // 300 nm
b.ConvertedTo(Micrometer).ToString(); // 0.03 µm

var c =a+b;
c.ToString(); // 0.03000003 m
c.ConvertedTo(Micrometer).ToString(); // 30000.03 µm
```
In the example, I didn't create millimeter nor micrometer, they are implicitely created when I declare meter main unit. Each amount keep its unit.
All the subUnits should be created when we declare a main unit (with a conditionnal boolean to be able to desactivate for custom units where metric prefix are irrelevants, metrix are relatives to SI units at least)

**Comment by petermorlion** [comment]
Interesting point. But due to the current implementation, this isn't possible. I can't dynamically add members to the static `LengthUnits` class. Also, it wouldn't make sense to automatically add these to certain units (e.g. there is no MilliYard or MilliFoot). Not all units in this library are metric  units.

I could add some missing things like nanometer and micrometer, but I fear I won't be able to cover all cases. The best option here, I think, is to register the extra units you need in your project.

**Comment by uncle-buzz** [comment]
Not all units are concerned by metrix prefix, so it's why It would be conditionnal. Even a subUnit, as a unit wouldn't create subsubUnit...
As you pointed it, the static implementation don't let this system to be added.
So, a regex correction on the source code on the few units concerned could be used to update the library, custom units wouldn't be concerned if all SI units are already presents int the library.
Thanks to pointed this.

**Comment by petermorlion** [comment]
Ah okk, I think I understand now. You would just like all metric prefixes to be available? Not only the few that are present now? I can easily add that.

**Comment by uncle-buzz** [comment]
Yes, but since units with prefix are always build the same way, I though about a programaticaly way to build them all, but I missed the static point.

**Comment by petermorlion** [comment]
Come to think of it, I don't think I will be adding all of them. They wouldn't all be used very often. For example, the usage scenario of a yottameter is niche to say the least. In such a case, I believe the developers should register this themselves. But I will add several that I think make sense.

**Comment by petermorlion** [comment]
I've added some new units in [version 3.0.0](https://github.com/petermorlion/RedStar.Amounts/releases/tag/3.0.0). I fixed a typo in [version 3.0.1](https://github.com/petermorlion/RedStar.Amounts/releases/tag/3.0.1), so best to use that one. I will upload a Nuget package shortly.
